apps/token/src/routes/proposals/components/propose/proposal-form-download-json.tsx

```diff
@@ -0,0 +1,19 @@
+import { useTranslation } from 'react-i18next';
+import { Button } from '@vegaprotocol/ui-toolkit';
+
+interface ProposalFormDownloadJsonProps {
+  downloadJson: () => void;
+}
+
+export const ProposalFormDownloadJson = ({
```

Comment:
question (non-blocking): Why not just have this component take a blob of data and stringify and do the download logic itself rather than a function?

Reply:
It seemed better to me for this to be as dumb as possible, and for the download logic to be a separate entity, which could be used independently. I don't feel strongly though, I can adjust if you think it's preferable to be in the component